contracts/core/Allo.sol

```diff
@@ -285,9 +253,10 @@ contract Allo is Transfer, Initializable, Ownable, AccessControl {
 
         Pool memory pool = Pool({
             identityId: _identityId,
-            allocationStrategy: _allocationStrategy,
-            distributionStrategy: _distributionStrategy,
+            strategy: _strategy,
             metadata: _metadata,
+            token: _token,
+            amount: 0, // this value is updated in _fundPool
```

Comment:
we don't need token and amount here, since these values are stored on the strategy
these values SHOULD always be in sync, but we cannot guarantee this, as we also (eventually) allow custom strategies.

i would propose to have a single source of truth approach here and remove it from the struct and fully rely on the strategy.

Reply:
It looks like we want to store it on Pool struct. I quite like that logic. The skirt logic works with this as well.